api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java

```diff
@@ -210,6 +211,17 @@ public ResponseEntity<PaymentDto> retrievePaymentStatus(@PathVariable("paymentRe
         return new ResponseEntity<>(creditAccountDtoMapper.toRetrievePaymentStatusResponse(payment), HttpStatus.OK);
     }
 
+    @ApiOperation(value = "Delete credit account payment details by payment reference", notes = "Delete payment details for supplied payment reference")
+    @ApiResponses(value = {
+            @ApiResponse(code = 204, message = "Payment deleted successfully"),
+            @ApiResponse(code = 404, message = "Payment not found for the given reference")
+    })
+    @RequestMapping(value = "/credit-account-payments/{paymentReference}", method = DELETE)
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteByPaymentReference(@PathVariable("paymentReference") String paymentReference) {
+        creditAccountPaymentService.deleteByPaymentReference(paymentReference);
+    }
+
```

Comment:
Address the warnings for the Controller Code that is not addressed...

Reply:
Updated code

Reply:
Resolved.